pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -556,11 +556,16 @@ func (c *Controller) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 			Fn:           flow.TaskFn(botanist.DeleteSeedNamespace).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(syncPoint, deleteDNSProviders, destroyReferencedResources),
 		})
-		_ = g.Add(flow.Task{
+		waitUntilNamespaceDeleted = g.Add(flow.Task{
 			Name:         "Waiting until shoot namespace in Seed has been deleted",
 			Fn:           botanist.WaitUntilSeedNamespaceDeleted,
 			Dependencies: flow.NewTaskIDs(deleteNamespace),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Deleting Shoot State",
+			Fn:           botanist.DeleteShootState,
+			Dependencies: flow.NewTaskIDs(syncPoint, waitUntilNamespaceDeleted),
```

Comment:
```suggestion
			 Dependencies: flow.NewTaskIDs(waitUntilNamespaceDeleted),
```
Dependency to `syncPoint` is implicitly given already.

As said in person, I don't have a strong opinion on when to do add this step in the flow, but I think, doing it in parallel to `WaitUntilSeedNamespaceDeleted` is also fine.

Reply:
Yep, think it would work if you do it in parallel to `WaitUntilSeedNamespaceDeleted` and just add a dependency to `deleteNamespace`